component/src/test/java/org/wso2/extension/siddhi/io/prometheus/source/SourceValidationTestcase.java

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.prometheus.source;
+
+import org.apache.log4j.Logger;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.extension.siddhi.io.prometheus.util.PrometheusConstants;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+
+/**
+ * Test cases for invalid source definitions.
+ */
+public class SourceValidationTestcase {
+    private static final Logger log = Logger.getLogger(SourceValidationTestcase.class);
+    private static String targetURL;
+    private static final String ERROR_MESSAGE = "Error on \'(.*)\' @ Line: (.*). Position: (.*), near \'(.*)\'. ";
+
+
+    @BeforeClass
+    public static void startTest() {
+        log.info("== Prometheus source validation tests started ==");
+        targetURL = "http://localhost:9080";
+    }
+
+    @AfterClass
+    public static void shutdown() throws InterruptedException {
+        Thread.sleep(100);
+        log.info("== Prometheus source validation tests completed ==");
+    }
+
+
+    private void startSiddhiApp(String streamDefinition) {
+        SiddhiManager siddhiManager = new SiddhiManager();
+        String siddhiApp = "@App:name('TestSiddhiApp')";
+        String outputStream = " @sink(type='log', prefix='test')" +
+                "define stream OutputStream (metric_name String, metric_type String, help String," +
+                " name String, age String, subtype String, le String, value double);";
+        String query = (
+                "@info(name = 'query') "
+                        + "from SourceTestStream "
+                        + "select * "
+                        + "insert into OutputStream;"
+        );
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(siddhiApp + streamDefinition +
+                outputStream + query);
+        InputHandler inputStream = siddhiAppRuntime.getInputHandler("SourceTestStream");
+        siddhiAppRuntime.start();
```

Comment:
Since the siddhiAppRuntime is not necessary to start for validation test cases, I removed the redundant lines in 7c67e1b27a3297df90f5e0acc96a251d2fcef4b5